cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs

```diff
@@ -0,0 +1,335 @@
+//! Plugin which implements fetching a Cincinnati graph via HTTP from a `/v1/graph`-compliant endpoint.
+//!
+//! Instead of processing the input graph, ghis plugin fetches a graph from a
+//! remote endpoint, which makes it effectively discard any given input graph.
+
+use crate::plugins::{
+    AsyncIO, BoxedPlugin, InternalIO, InternalPlugin, InternalPluginWrapper, PluginSettings,
+};
+use crate::CONTENT_TYPE;
+use commons::GraphError;
+use failure::Fallible;
+use futures::{future, Future, Stream};
+use prometheus::{Counter, Registry};
+use reqwest;
+use reqwest::header::{HeaderValue, ACCEPT};
+
+/// Default URL to upstream graph provider.
+pub static DEFAULT_UPSTREAM_URL: &str = "http://localhost:8080/v1/graph";
+
+/// Plugin settings.
+#[derive(Clone, CustomDebug, Deserialize, SmartDefault)]
+#[serde(default)]
+struct CincinnatiGraphFetchSettings {
+    #[default(DEFAULT_UPSTREAM_URL.to_string())]
+    upstream: String,
```

Comment:
Side note: we talked about splitting this into a `base_url` + `api_endpoint` later on.